gridstatus/ieso.py

```diff
@@ -0,0 +1,473 @@
+            days=MAXIMUM_DAYS_IN_PAST_FOR_LOAD,
+        ):
+            raise NotSupported(
+                f"Load data is not available for dates more than "
+                f"{MAXIMUM_DAYS_IN_PAST_FOR_LOAD} days in the past.",
+            )
+
+        # Don't request data from after now because it does not exist
+        if end:
+            end = min(pd.Timestamp.now(tz=self.default_timezone), end)
+
+        return self._retrieve_5_minute_load(date, end, verbose)
+
+    @support_date_range(frequency="HOUR_START")
+    def _retrieve_5_minute_load(self, date, end=None, verbose=False):
+        # We have to add 1 to the hour to get the file because the filename with
+        # hour x contains data for hour x-1. For example, to get data for
+        # 9:00 - 9:55, we need to request the file for hour 10.
+        # The hour should be in the range 1-24
+        hour = date.hour + 1
+
+        url = LOAD_TEMPLATE_URL.replace(
+            "YYYYMMDDHH",
+            f"{(date).strftime('%Y%m%d')}{hour:02d}",
+        )
+
+        r = self._request(url, verbose)
+
+        root = ET.fromstring(r.text)
+
+        # Extracting all triples of Interval, Market Total Load, and Ontario Load
+        interval_loads_and_demands = self._find_loads_at_each_interval_from_xml(root)
+
+        df = pd.DataFrame(
+            interval_loads_and_demands,
+            columns=["Interval", "Market Total Load", "Ontario Load"],
+        )
+
+        delivery_date = root.find("DocBody/DeliveryDate", NAMESPACES_FOR_XML).text
+        delivery_hour = int(root.find("DocBody/DeliveryHour", NAMESPACES_FOR_XML).text)
+        created_at = root.find("DocHeader/CreatedAt", NAMESPACES_FOR_XML).text
+
+        df["Delivery Date"] = pd.Timestamp(delivery_date, tz=self.default_timezone)
+
+        # The starting hour is 1, so we subtract 1 to get the hour in the range 0-23
+        df["Delivery Hour Start"] = delivery_hour - 1
+        # Multiply the interval minus 1 by 5 to get the minutes in the range 0-55
+        df["Interval Minute Start"] = MINUTES_INTERVAL * (df["Interval"] - 1)
+        df["Published Time"] = pd.Timestamp(created_at, tz=self.default_timezone)
+
+        df["Interval Start"] = (
+            df["Delivery Date"]
+            + pd.to_timedelta(df["Delivery Hour Start"], unit="h")
+            + pd.to_timedelta(df["Interval Minute Start"], unit="m")
+        )
+
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(
+            minutes=MINUTES_INTERVAL,
+        )
+
+        cols_to_keep_in_order = [
+            "Interval Start",
+            "Interval End",
+            "Published Time",
+            "Market Total Load",
+            "Ontario Load",
+        ]
+
+        df = utils.move_cols_to_front(df, cols_to_keep_in_order)[cols_to_keep_in_order]
+
+        if end:
+            return df[df["Interval End"] <= pd.Timestamp(end)]
+
+        return df
+
+    def get_load(self, date, end=None, verbose=False):
```

Comment:
let's do just one method for `get_load` and `get_5_min_load` since they are using the same underlying data all processing should be same except for the final resample

something like `get_load(frequency="5min")` or `get_load(frequency="hourly")`. i would default to "5min"

Reply:
Adding `frequency` changes the interface pattern to `get_load` only for this ISO. Generally, I'm in favor of maintaining a consistent external interface to make it easier for users. 

I can combine them into one function, but I think they are better as two methods.

Reply:
I agree with that, but my view is If you set a default value for the extra params, then a caller can use this method just like any of the others, so it doesn’t violate things. But if they want to take advantage of the optional argument they can.

I just think it’s potentially confusing to have a second method that has the same underlying data. By combining into one and giving the optional parameter we make it more explicit we are just doing processing on the raw data

Reply:
Thanks for explaining your reasoning. I've updated the PR to have a single `get_load` method that supports "5min" and "H" frequencies (we could support any frequency supported by Pandas if we used `resample`).